Protocol.md

```diff
@@ -946,20 +946,21 @@ authenticate itself towards the initiator.
 ## Message States
 
 ```
-         +-------+    +-----------------+    +------------------+
-    --+->+ token |    | key (initiator) +--->+ auth (responder) |
-      |  +---+---+    +--------+--------+    +---------+--------+
-      |      |                 ^                       |
-      |      |                 |                       v
-      |      |        +--------+--------+    +---------+--------+
-      +------+------->+ key (responder) |    | auth (initiator) |
-                      +-----------------+    +---+----------+---+
-                                                 :          |
-                                                 :          v
-                                                 v      +---+---+
-                                                        | close |
-                                               Task     +-------+
-
+         +-----------------+    +------------------+
+         | key (initiator) +--->+ auth (responder) |
+         +--------+--------+    +---------+--------+
+                  ^                       |
+                  |                       v
+         +--------+--------+    +---------+--------+
+    --+->+ key (responder) |    | auth (initiator) +.....> Task
+      |  +---+-------------+    +--+-----------+---+  :
+      |      ^                     |           |      :
+      |      |                     v           v      :
+      |  +---+---+      +----------+--+    +---+---+  :
+      +->+ token |      | application +--->+ close |  :
+         +-------+      +----------+--+    +-------+  :
+                                   :                  :
+                                   :..................:
```

Comment:
Because a 'close' message can be sent after 'application' and 'auth' before the task has a chance to kick in. For example, if no task could be negotiated.